eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -380,3 +384,48 @@ case object EmptyAPIPasswordException extends RuntimeException("must set a passw
 case object IncompatibleDBException extends RuntimeException("database is not compatible with this version of eclair")
 
 case object IncompatibleNetworkDBException extends RuntimeException("network database is not compatible with this version of eclair")
+
+object Setup {
+  def reconcileWatchAddresses(bitcoinClient: ExtendedBitcoinClient, databases: Databases)(implicit ec: ExecutionContext, logger: Logger): Future[Unit] = {
+    // lookup existing addresses in bitcoin
+    bitcoinClient.listReceivedByAddress(minConf = 1, includeEmpty = true, includeWatchOnly = true).map(_.filter(_.label == "IMPORTED")).flatMap {
+      case Nil =>
```

Comment:
This fails to import channels in a brand new bitcoind instance if it has zero addresses already imported.